core/algorithms/onpolicy_sync/engine.py

```diff
@@ -1261,7 +1261,7 @@ def __init__(
         deterministic_cudnn: bool = False,
         mp_ctx: Optional[BaseContext] = None,
         device: Union[str, torch.device, int] = "cpu",
-        deterministic_agent: bool = True,
+        deterministic_agent: bool = False,
```

Comment:
why is it better to have this default?

Reply:
currently there's no code in runner to change the agent type, so we'll use sampling for `train`, `valid` and `test`. do we always want this?

Reply:
I think having it `False` is, somewhat surprisingly, actually more robust. For instance, I just ran Klemen's debug model and with a deterministic agent it just continuously runs into a wall while the probabilistic variant manages to find the object semi-frequently.

Reply:
We should add a command line flag to allow for deterministic agents. I'll add an issue.